src/Microsoft.Graph.Core/Requests/BaseRequest.cs

```diff
@@ -130,6 +130,7 @@ public class BaseRequest : IBaseRequest
                 if (response.Content != null)
                 {
                     var responseString = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                    responseString = AddHeadersToResponse(response, responseString);
```

Comment:
Instead of pass the responseString, and the response (which has the response content as well), I suggest that we change this call from AddHeadersToResponse to `string GetResponseString(HttpResponseMessage hrm)`. GetResponseString will  call AddHeadersToResponse with the results of response.Content.ReadAsStringAsync() and then return the results.

Reply:
That makes a lot more sense